### PR TITLE
PWN - 7168 - swap widget, fragment xml

### PR DESCRIPTION
**Pull request by NikolaiEmelianov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN - 7168

## Description of Work

swap widget, fragment xml

## Screenshots (Optional)

photo_2023-02-16 17 27 55

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidget.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    fun bind(model: SwapWidgetModel) {
+        when (model) {
+            is SwapWidgetModel.Content -> bindContent(model)
+            is SwapWidgetModel.Loading -> bindLoading(model)
+        }
+    }
+
+    private fun bindLoading(model: SwapWidgetModel.Loading) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidget.kt:43]: private fun bindLoading(model: SwapWidgetModel.Loading) = with(binding) {
and get rid of binding on every line

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidget.kt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        binding.textViewWidgetTitle.bindOrGone(model.widgetTitle)
+        binding.textViewAvailableAmountTitle.isVisible = false
+        binding.textViewAvailableAmountValue.isVisible = false
+        binding.textViewCurrencyName.bindOrGone(model.currencySkeleton)
+        bindInput(model, model.amountSkeleton)
+        binding.textViewBalance.bindOrGone(model.balanceSkeleton)
+        binding.textViewFiatAmount.isVisible = false
+    }
+
+    private fun bindContent(model: SwapWidgetModel.Content) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidget.kt:54]: same here and in other places

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        binding.imageViewSwapTokens.background = shapeDrawable(shapeCircle())
+        binding.imageViewSwapTokens.backgroundTintList = view.context.getColorStateList(R.color.button_rain)
+        binding.imageViewSwapTokens.rippleForeground(shapeCircle())
+        binding.imageViewSwapTokens.setOnClickListener { }
+        setYellowAller()
+        binding.linearLayoutAlert.isVisible = true
+    }
+
+    private fun setYellowAller() {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt:61]: Aller -> Alert?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        val context = binding.root.context
+        DrawableCellModel(
+            drawable = shapeDrawable(shapeRoundedAll(8f.toPx())),
+            tint = R.color.bg_light_sun,
+            strokeWidth = 1f.toPx(),
+            strokeColor = R.color.bg_sun,
+        ).applyBackground(binding.linearLayoutAlert)
+        binding.imageviewAlert.imageTintList = context.getColorStateList(R.color.icons_sun)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt:69]: imageViewAlert

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    width = 160.toPx(),
+                    radius = 4f.toPx(),
+                )
+            )
+        )
+
+        binding.imageViewSwapTokens.background = shapeDrawable(shapeCircle())
+        binding.imageViewSwapTokens.backgroundTintList = view.context.getColorStateList(R.color.button_rain)
+        binding.imageViewSwapTokens.rippleForeground(shapeCircle())
+        binding.imageViewSwapTokens.setOnClickListener { }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt:56]: let's leave todo with link to the task

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidgetModel.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    data class Loading(
+        val isStatic: Boolean = false,
+        val widgetTitle: TextViewCellModel.Raw,
+        val currencySkeleton: TextViewCellModel.Skeleton = TextViewCellModel.Skeleton(bigSkeleton()),
+        val amountSkeleton: TextViewCellModel.Skeleton = TextViewCellModel.Skeleton(bigSkeleton()),
+        val balanceSkeleton: TextViewCellModel.Skeleton = TextViewCellModel.Skeleton(smallSkeleton()),
+    ) : SwapWidgetModel
+}
+
+fun swapToAmountLoading(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/widget/SwapWidgetModel.kt:29]: better place as a method for  Content class to remove ambiguity in autocomplete and search

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**